core/src/main/java/org/fao/geonet/kernel/DataManager.java

```diff
@@ -122,24 +122,33 @@ public static void setNamespacePrefix(final Element md) {
     }
 
     /**
-     * Init Data manager and refresh index if needed. Can also be called after GeoNetwork startup in order to rebuild the lucene index
+     * Init data manager components.
      *
-     * @param force Force reindexing all from scratch
+     * @param context Service context used for setup
      **/
-    public void init(ServiceContext context, Boolean force) throws Exception {
-        this.metadataIndexer.init(context, force);
-        this.metadataManager.init(context, force);
-        this.metadataUtils.init(context, force);
+    public void init(ServiceContext context) throws Exception {
+        this.metadataIndexer.init(context);
+        this.metadataManager.init(context);
+        this.metadataUtils.init(context);
 
         // FIXME this shouldn't login automatically ever!
-        if (context.getUserSession() == null) {
-            LOGGER_DATA_MANAGER.debug("Automatically login in as Administrator. Who is this? Who is calling this?");
-            UserSession session = new UserSession();
-            context.setUserSession(session);
-            session.loginAs(new User().setUsername("admin").setId(-1).setProfile(Profile.Administrator));
-            LOGGER_DATA_MANAGER.debug("Hopefully this is cron job or routinely background task. Who called us?",
-                new Exception("Dummy Exception to know the stacktrace"));
-        }
+//        if (context.getUserSession() == null) {
+//            LOGGER_DATA_MANAGER.debug("Automatically login in as Administrator. Who is this? Who is calling this?");
+//            UserSession session = new UserSession();
+//            context.setUserSession(session);
+//            session.loginAs(new User().setUsername("admin").setId(-1).setProfile(Profile.Administrator));
+//            LOGGER_DATA_MANAGER.debug("Hopefully this is cron job or routinely background task. Who called us?",
+//                new Exception("Dummy Exception to know the stacktrace"));
+//        }
```

Comment:
This could should just be removed, or if we like on main a failure could be raised.
```
if (context.getUserSession() == null) {
   throw new IllegalStateException( "Anonymous data manager access not available");
}
```

Reply:
I was being so careful not to break existing workflow I did not introduce any additional strict checking.